edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Docker/CombinedEdgeletConfigProvider.cs

```diff
@@ -29,8 +29,13 @@ public override CombinedDockerConfig GetCombinedConfig(IModule module, IRuntimeI
         {
             CombinedDockerConfig combinedConfig = base.GetCombinedConfig(module, runtimeInfo);
 
-            // if the workload URI is a Unix domain socket then volume mount it into the container
             CreateContainerParameters createOptions = CloneOrCreateParams(combinedConfig.CreateOptions);
+
+            // before making any other modifications to createOptions, save edge agent's createOptions + env as
+            // container labels so they're available as soon as it loads
+            InjectEdgeAgentLabels(module, createOptions);
```

Comment:
Should we name it as "InjectContainerLabels"? Currently I am confused why we need to call InjectEdgeAgentLabels even it is creating other modules.

Reply:
Because it only injects the labels for edgeAgent. See the implementation of the method:
```
if (module.Name.Equals(Constants.EdgeAgentModuleName, StringComparison.OrdinalIgnoreCase))...
```